src/test/scala/ce3/laws/AsyncRegionTests.scala

```
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ce3
+package laws
+
+import cats.{Eq, Group, Order}
+import cats.data.EitherT
+import cats.laws.discipline._
+import cats.laws.discipline.SemigroupalTests.Isomorphisms
+
+import org.scalacheck._, Prop.forAll
+import org.scalacheck.util.Pretty
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+
+trait AsyncRegionTests[R[_[_], _], F[_]] extends AsyncTests[R[F, ?]] with TemporalRegionTests[R, F, Throwable] {
+
+  val laws: AsyncRegionLaws[R, F]
+
+  def asyncRegion[A: Arbitrary: Eq, B: Arbitrary: Eq, C: Arbitrary: Eq](tolerance: FiniteDuration)(
+    implicit
+      ArbRFA: Arbitrary[R[F, A]],
```

I really wonder how you're debugging diverging expansions in these...

---

Ah! That I can answer: I ignore the error. It seems that NSC just defaults to "diverging implicit expansion" whenever it's trying to resolve a zillion things and any of them fails. So I debug the errors by manually materializing things that I expect might be problematic until I find one that doesn't materialize, mostly because I haven't defined it. I've only found one case where I *really* had to help the compiler along (the `Tolerance` implicit).

---

Yeah, that's pretty much how I do it... and it hurts!